File: src/201-300/268. Missing Number/Solution.js
// More Efficient Solution
/**
 * @param { number[] } nums
 * @return { number }
 */
const missingNumber = function (nums) {
	return ((nums.length * (nums.length + 1)) / 2) - nums.reduce((acc, cur) => acc + cur, 0)
}


// Less Efficient Solution
/**
 * @param { number[] } nums
 * @return { number }
 */
const missingNumber2 = function (nums) {
	let res = nums.length

	for (let i = 0; i < nums.length; i++) {
		res -= nums[i] - i
	}

	return res
}

File: src/201-300/268. Missing Number/Solution.py
from typing import List


# More Efficient Solution
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        return int((len(nums) * (len(nums) + 1)) / 2) - sum(nums)


# Less Efficient Solution
class Solution2:
    def missingNumber(self, nums: List[int]) -> int:
        res = len(nums)

        for i in range(len(nums)):
            res -= nums[i] - i

        return res
